Project Closeout Report

SuganyaAK opened the pull request:
Production grade Dapps - Final closeout report 
---
I have created a draft report based on the template shared, I included the challenge KPIs and Project KPIs based on the documentation we have and the milestones achieved so far.

---
I would appreciate your feedback, review comments and suggestions.

1. **Project Manager** - Please confirm who should be listed as the "Project Manager"
2. **Project Completion Date** - Do we have to include the completion date close to our submission , or should I use the estimated milestone date on our proposal?
3. In our final milestone output, it is mentioned **"Collaboration with Demeter where Smart Contracts are available to launch in their platform"**,What should our next steps be regarding this?
4. In this document, under "Adoption Rate and Usage Metrics," I have included the GitHub insights for the metrics. Is this correct?

SuganyaAK committed:
Project Closeout Report - Draft copy

SuganyaAK committed:
Updated the font to "Computer Modern"

SuganyaAK committed:
Updated "link" for Adoption Rate and Usage Metrics

nikhils9 commented on pages/closeoutreport.typ:
```diff
@@ -0,0 +1,55 @@
+
+/ List of challenge KPIs and how the project addressed them :
+
+- *Addressing the Scarcity : * Aimed to reduce the scarcity of openly available, production-level codebases, the project achieved its goal by providing a suite of five versatile, open-source smart contract libraries. These libraries are readily accessible and can be adopted for general use by removing application-specific logic.
```
I think we can rephrase this 
> These libraries are readily accessible and can be adopted for general use by removing application-specific logic.

as
These libraries are readily accessible and can be adopted for general use as it is or by incorporating additional application-specific logic.

SuganyaAK replied:
Revised and modified.

nikhils9 commented on pages/closeoutreport.typ:
```diff
@@ -0,0 +1,55 @@
+
+- *Addressing the Scarcity : * Aimed to reduce the scarcity of openly available, production-level codebases, the project achieved its goal by providing a suite of five versatile, open-source smart contract libraries. These libraries are readily accessible and can be adopted for general use by removing application-specific logic.
+
+- *Fostering Ecosystem Innovation :* The five smart contract libraries are a testament to best practices in smart contract design, testing, and optimization. Designed with composability in mind, they serve as foundational blocks for building complex DApps. These libraries are ideal for teams aspiring to transition or start fresh in the Cardano ecosystem, streamlining the development process and f ...
```
```suggestion
- *Fostering Ecosystem Innovation :* The five smart contract libraries are a testament to best practices in smart contract design, testing and optimization. Designed with composability in mind, they serve as foundational blocks for building complex DApps. These libraries are ideal for teams aspiring to transition or start fresh in the Cardano ecosystem, streamlining the development process and fostering innovation within the ecosystem.
```

SuganyaAK replied:
Updated the suggested change.

nikhils9 commented on pages/closeoutreport.typ:
```diff
@@ -0,0 +1,55 @@
+
+- *Fostering Ecosystem Innovation :* The five smart contract libraries are a testament to best practices in smart contract design, testing, and optimization. Designed with composability in mind, they serve as foundational blocks for building complex DApps. These libraries are ideal for teams aspiring to transition or start fresh in the Cardano ecosystem, streamlining the development process and f ...
+
+- *Ensure Code Quality and Production-Ready Resources : * The challenge was to uphold high standards of code quality, adherence to best practices, and readiness for production, aimed at alleviating the pain points faced by developers. The quality of the smart contract libraries was assessed through comprehensive methods, including code reviews, unit testing, and property-based testing.
```
```suggestion
- *Ensure Code Quality and Production-Ready Resources : * The challenge was to uphold high standards of code quality, adherence to best practices, and readiness for production, aimed at alleviating the pain points faced by developers. The quality of the smart contract libraries was assessed through comprehensive methods, including code reviews, unit testing and property-based testing.
```

SuganyaAK replied:
Updated.

nikhils9 commented on pages/closeoutreport.typ:
```diff
@@ -0,0 +1,55 @@
+- *Extensive Documentation  :* Comprehensive documentation for each smart contract library, including detailed explanations of contract functionality, parameters, and usage, was provided. Detailed diagrams aid in understanding contract architecture and data flows.
+
+
+/ Key achievements (in particular around collaboration and engagement) :
```
Maybe the proposal deliverables can be re-stated here as achievements along with necessary collaboration with Demeter team.

MladenLm replied:
Yup, let's go with that

nikhils9 commented:
> ## Production grade Dapps - Final closeout report
> 
> I have created a draft report based on the template shared, I included the challenge KPIs and Project KPIs based on the documentation we have and the milestones achieved so far.
> 
> I would appreciate your feedback, review comments and suggestions.
> 
>     1. **Project Manager** - Please confirm who should be listed as the "Project Manager"

Jonathan?

> 
>     2. **Project Completion Date** - Do we have to include the completion date close to our submission , or should I use the estimated milestone date on our proposal?

Not sure.

> 
>     3. In our final milestone output, it is mentioned **"Collaboration with Demeter where Smart Contracts are available to launch in their platform"**,What should our next steps be regarding this?

I think we will need to achieve this goal before submitting to catalyst reviewers.
> 
>     4. In this document, under "Adoption Rate and Usage Metrics," I have included the GitHub insights for the metrics. Is this correct?

Yes, it seems to be a good solution. However, since this repository is just a Readme with list of actual project repository links, it would be better to provide the links to individual repo's insights.

@MladenLm can you please provide your inputs/ suggestions on the open questions here.

nikhils9 commented:
I'll try answering the below sections expecting improvements/ reviews from others in order to finalize.

/ Key learnings:
- Enhanced understanding of different business domains in which these smart contract projects provide value.
- Improvements in tooling and style of documentation to facilitate easy onboarding of developers.

/ Next steps for the product or service developed:
- Expansion by adding new production grade dapps
- Increasing adoption by making these smart contracts available in different smart contract languages.

Please try and elaborate on above if feasible.

SuganyaAK commented:
> > ## Production grade Dapps - Final closeout report
> > I have created a draft report based on the template shared, I included the challenge KPIs and Project KPIs based on the documentation we have and the milestones achieved so far.
> > I would appreciate your feedback, review comments and suggestions.
> > ```
> > 1. **Project Manager** - Please confirm who should be listed as the "Project Manager"
> > ```
> 
> Jonathan?
> 
> > ```
> > 2. **Project Completion Date** - Do we have to include the completion date close to our submission , or should I use the estimated milestone date on our proposal?
> > ```
> 
> Not sure.
> 
> > ```
> > 3. In our final milestone output, it is mentioned **"Collaboration with Demeter where Smart Contracts are available to launch in their platform"**,What should our next steps be regarding this?
> > ```
> 
> I think we will need to achieve this goal before submitting to catalyst reviewers.
> 
> > ```
> > 4. In this document, under "Adoption Rate and Usage Metrics," I have included the GitHub insights for the metrics. Is this correct?
> > ```
> 
> Yes, it seems to be a good solution. However, since this repository is just a Readme with list of actual project repository links, it would be better to provide the links to individual repo's insights.
> I linked to each project's insights as suggested..

> @MladenLm can you please provide your inputs/ suggestions on the open questions here.

SuganyaAK committed:
Incorporated review comments on key learnings and next steps

SuganyaAK commented:
> I'll try answering the below sections expecting improvements/ reviews from others in order to finalize.
> 
> / Key learnings:
> 
> * Enhanced understanding of different business domains in which these smart contract projects provide value.
> * Improvements in tooling and style of documentation to facilitate easy onboarding of developers.
> 
> / Next steps for the product or service developed:
> 
> * Expansion by adding new production grade dapps
> * Increasing adoption by making these smart contracts available in different smart contract languages.
> 
> Please try and elaborate on above if feasible.

I have elaborated based on your inputs. Kindly review. However, as per your expectations will keep it open to include any other comments/suggestions from the team.

nikhils9 commented:
> > I'll try answering the below sections expecting improvements/ reviews from others in order to finalize.
> > / Key learnings:
> > 
> > * Enhanced understanding of different business domains in which these smart contract projects provide value.
> > * Improvements in tooling and style of documentation to facilitate easy onboarding of developers.
> > 
> > / Next steps for the product or service developed:
> > 
> > * Expansion by adding new production grade dapps
> > * Increasing adoption by making these smart contracts available in different smart contract languages.
> > 
> > Please try and elaborate on above if feasible.
> 
> I have elaborated based on your inputs. Kindly review. However, as per your expectations will keep it open to include any other comments/suggestions from the team.

Looks good to me!

solidsnakedev commented:
> > ## Production grade Dapps - Final closeout report
> > I have created a draft report based on the template shared, I included the challenge KPIs and Project KPIs based on the documentation we have and the milestones achieved so far.
> > I would appreciate your feedback, review comments and suggestions.
> > ```
> > 1. **Project Manager** - Please confirm who should be listed as the "Project Manager"
> > ```
> 
> Jonathan?
> 
> > ```
> > 2. **Project Completion Date** - Do we have to include the completion date close to our submission , or should I use the estimated milestone date on our proposal?
> > ```
> 
> Not sure.
> 
> > ```
> > 3. In our final milestone output, it is mentioned **"Collaboration with Demeter where Smart Contracts are available to launch in their platform"**,What should our next steps be regarding this?
> > ```
> 
> I think we will need to achieve this goal before submitting to catalyst reviewers.
> 
> > ```
> > 4. In this document, under "Adoption Rate and Usage Metrics," I have included the GitHub insights for the metrics. Is this correct?
> > ```
> 
> Yes, it seems to be a good solution. However, since this repository is just a Readme with list of actual project repository links, it would be better to provide the links to individual repo's insights.
> 
> @MladenLm can you please provide your inputs/ suggestions on the open questions here.

yes, you can add me as project manager @SuganyaAK

SuganyaAK commented:
@MladenLm @solidsnakedev

I understand the importance of including a cover page with logo in a audit report. 
However, considering the closeout report guidelines specify a 2-3 page written summary, 
how crucial is it to add the logo and contents page in the closeout report?

Appreciate your thoughts on this.

![closeoutreport_guidelines](https://github.com/Anastasia-Labs/production-grade-dapps/assets/99989138/9b79610e-0094-41bb-bdf3-d87dd2e11f0d)

MladenLm commented:
I think 3 pages is not a hard limit, and I would like to have our branding/logo on the report so it can be recognizable when shared in the community

MladenLm commented:
1. Please put Jonathan as a project manager
2. Date of completion can be May 31th 
3. Hayden published our contracts on Demeter just today - please link those in the report too - https://demeter.run/ports, https://demeter.run/starter-kits/anastasialabs-linear-vesting
4. That is good. In case we will have to add something, the Catalyst team will tell us

SuganyaAK committed:
Includes formmating and content for all titles under report guidelines

SuganyaAK commented:
I have updated the content according to your comments and adjusted the page layout to better suit the needs of this report. Please review and let me know if any further suggestions or modifications are needed.

SuganyaAK committed:
Added a closeout report folder
